### ISSUE TITLE
systemd-udevd: Failed to get link config: No such device, Could not generate persistent MAC: No data available

### ISSUE THREAD

**Issue by reynico** [systemd/systemd#19428]
**systemd version the issue has been seen with**
 > systemd 245 (245.4-4ubuntu3.5)

**Used distribution**
 > Ubuntu 20.04.2 LTS

**Linux kernel version used** (`uname -a`)
 > `Linux ip-10-10-1-48 5.9.12-050912-generic #202012020835 SMP Wed Dec 2 09:10:22 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > Create virtual interface

**Unexpected behaviour you saw**
 > wu8b300p9lf3-0: Failed to get link config: No such device, wu8b300p9lf3-1: Could not find device, waiting for device initialization: No such device

**Steps to reproduce the problem**
 > We run a CI pipeline with LXC containers using Concourse

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.700255307Z","level":"info","source":"guardian","message":"guardian.create.network.config-create","data":{"config":{"ContainerHandle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","HostIntf":"wu8b300p9lf3-0","ContainerIntf":"wu8b300p9lf3-1","IPTablePrefix":"w--","IPTableInstance":"u8b300p9lf3","BridgeName":"wbrdg-0afe0050","BridgeIP":"10.254.0.81","ContainerIP":"10.254.0.82","ExternalIP":"10.10.1.48","Subnet":{"IP":"10.254.0.80","Mask":"/////A=="},"Mtu":1500,"PluginNameservers":null,"OperatorNameservers":["10.10.0.2"],"AdditionalNameservers":[],"AdditionalHostEntries":null,"PluginSearchDomains":null},"handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.7","spec":""}}
Apr 26 07:17:12 ip-10-10-1-48 systemd-networkd[548]: rtnl: received neighbor for link '624' we don't know about, ignoring.
Apr 26 07:17:12 ip-10-10-1-48 systemd-udevd[510173]: Using default interface naming scheme 'v245'.
Apr 26 07:17:12 ip-10-10-1-48 networkd-dispatcher[605]: WARNING:Unknown index 624 seen, reloading interface list
Apr 26 07:17:12 ip-10-10-1-48 systemd-udevd[510173]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Apr 26 07:17:12 ip-10-10-1-48 systemd-udevd[510173]: wbrdg-0afe0050: Could not generate persistent MAC: No data available
Apr 26 07:17:12 ip-10-10-1-48 systemd-networkd[548]: wbrdg-0afe0050: Link UP
Apr 26 07:17:12 ip-10-10-1-48 systemd-udevd[510163]: wu8b300p9lf3-0: Failed to get link config: No such device
Apr 26 07:17:12 ip-10-10-1-48 networkd-dispatcher[605]: WARNING:Unknown index 625 seen, reloading interface list
Apr 26 07:17:12 ip-10-10-1-48 systemd-networkd[548]: wu8b300p9lf3-1: Could not find device, waiting for device initialization: No such device
Apr 26 07:17:12 ip-10-10-1-48 systemd-udevd[510156]: wu8b300p9lf3-1: Failed to get link config: No such device
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.747817408Z","level":"error","source":"guardian","message":"guardian.create.network.configure-host.veth.create","data":{"bridgeIP":"10.254.0.81","bridgeName":"wbrdg-0afe0050","containerIface":"wu8b300p9lf3-1","error":"devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","hostIface":"wu8b300p9lf3-0","mtu":1500,"pid":510284,"session":"23060.7.2.2","spec":"","subnet":{"IP":"10.254.0.80","Mask":"/////A=="}}}
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.747906837Z","level":"info","source":"guardian","message":"guardian.create.network.finished","data":{"handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.7","spec":""}}
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.747968921Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.start","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8"}}
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.747994034Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.destroy.started","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8.1"}}
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.748477793Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.destroy.delete.started","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8.1.1"}}
Apr 26 07:17:12 ip-10-10-1-48 networkd-dispatcher[605]: ERROR:Unknown interface index 625 seen even after reload
Apr 26 07:17:12 ip-10-10-1-48 networkd-dispatcher[605]: WARNING:Unknown index 625 seen, reloading interface list
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.782680908Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.destroy.delete.state","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8.1.1","state":{"Pid":510284,"Status":"running"}}}
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.782750325Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.destroy.delete.runc-delete.started","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8.1.1.2"}}
Apr 26 07:17:12 ip-10-10-1-48 networkd-dispatcher[605]: ERROR:Unknown interface index 625 seen even after reload
Apr 26 07:17:12 ip-10-10-1-48 concourse[811]: {"timestamp":"2021-04-26T07:17:12.892863290Z","level":"info","source":"guardian","message":"guardian.create.create-failed-cleaningup.destroy.delete.runc-delete.finished","data":{"cause":"network: failed to create veth pair with host interface name 'wu8b300p9lf3-0', container interface name 'wu8b300p9lf3-1': devices: create veth pair: operation not permitted","handle":"9e8b87cd-7230-429a-44f8-f633a4e0d11b","session":"23060.8.1.1.2"}}
```

Our CI system consists of three Ubuntu Linux - AWS EC2 machines. CI builds randomly fails with the error

```
network: failed to create veth pair with host interface name 'wtn50kbcfipc-0', container interface name 'wtn50kbcfipc-1': devices: create veth pair: operation not permitted
```

This happens only few times a week in all three machines/workers. All three machines has the same configuration, same hardware, same software version. The CI software (concourse) has a `retry` feature that retries each step. Sometimes, this workaround works as expected so we just see the `network: failed to create veth pair with host interface name` at the first try.

cloudfoundry/guardian is the software that manages the virtual interfaces, and the error comes from here: https://github.com/cloudfoundry/guardian/blob/b48a5e072ce3a7feabcbd93f31573abfeeba9c25/kawasaki/devices/veth_linux.go#L12

**Comment by poettering** [comment]
[Canned reply follows] This is the upstream bug and feature request tracker of systemd. Please use this only for issues in the two most current upstream systemd versions. See this link for the list of current releases.

https://github.com/systemd/systemd/releases

For support for older versions please contact your distribution instead.

In order to keep frustration at a minimum the submission forum tried to make this all very clear.

**Comment by poettering** [comment]
Long fixed in 3ec2f7f2e3fc04699d8de27d3cc2e5eecf9a3698
